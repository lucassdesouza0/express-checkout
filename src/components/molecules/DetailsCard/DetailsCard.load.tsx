import React, { useEffect, useState } from "react";
import Skeleton from "react-loading-skeleton";

import { Card, Details } from "./DetailsCard.styles";

const DetailsCard = () => {
  return (
    <Card>
      <Skeleton width={100} height={100} />
      <Details>
        <Skeleton />

        <Skeleton />
      </Details>
    </Card>
  );
};

export default DetailsCard;
