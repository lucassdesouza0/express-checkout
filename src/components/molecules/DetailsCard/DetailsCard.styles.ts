import styled from "@emotion/styled";

export const Card = styled("div")`
  display: flex;
  flex-flow: row wrap;
  flex-direction: row;
  overflow-y: auto;
  margin: 10px;
  padding: 10px;
  box-shadow: 2px 2px 5px 5px #eee;
`;

export const Details = styled("div")`
  flex: 1;
  margin-left: 20px;
`;

export const ProductPrice = styled("div")`
  padding: 10px;
  margin-left: 10px;
`;

export const Shipping = styled("div")`
  margin: 30px 0;

  p {
    margin: none;
    font-weight: 500;
  }
`;
