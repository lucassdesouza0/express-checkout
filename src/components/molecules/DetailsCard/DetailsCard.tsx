import React, { useContext } from "react";

import Image from "atoms/Image/Image";
import Price from "atoms/Price/Price";
import { ContextType, ProductContext } from "pages/_app";

import DetailsCardLoad from "./DetailsCard.load";

import { Card, Details, Shipping } from "./DetailsCard.styles";
import { useTranslation } from "next-i18next";
import { useRouter } from "next/dist/client/router";
import Button from "components/atoms/Button/Button";

const DetailsCard = () => {
  const { t } = useTranslation("common");
  const router = useRouter();
  const { productsContext, setProductsContext } = useContext(
    ProductContext
  ) as ContextType;

  const product = productsContext?.product;

  if (!product) return <DetailsCardLoad />;

  return (
    <Card>
      <Image
        src="https://images.prismic.io/barkyn/153c4252-8084-49b8-9931-3ec426cdd859_premium_box.png?auto=compress,format"
        width="100"
        height="100"
      />
      <Details>
        <p>{product?.name}</p>
        <Price value={product?.datail?.price} />
        <Shipping>
          {product?.shipping && (
            <>
              <p> {t("address")}</p>
              <ul>
                <li>{product?.shipping?.address}</li>
                <li>{product?.shipping?.postalCode}</li>
              </ul>
            </>
          )}
        </Shipping>
        {productsContext?.product?.shipping && !productsContext?.isComplete && (
          <Button type="button" onClick={() => router.push("success")}>
            {t("close")}
          </Button>
        )}
      </Details>
    </Card>
  );
};

export default DetailsCard;
