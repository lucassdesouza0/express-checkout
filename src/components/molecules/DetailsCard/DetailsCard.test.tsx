import React from "react";
import { screen, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import testRender from "../../../tests/render";
import ProductContext from "context/product";

import mock from "../../../tests/__mocks__/product";

import DetailsCard from "./DetailsCard";

describe("<ProductCard />", () => {
  const productsContext = { product: mock };
  test("it should mount", () => {
    testRender(<DetailsCard product={mock} />);

    const card = screen.getAllByTestId("details-card");

    expect(card[0]).toBeInTheDocument();
  });

  test("it should render the load version of the card", () => {
    const { asFragment } = render(<DetailsCard />);
    expect(asFragment()).toMatchSnapshot();
  });

  //   test("it should match snapshot", () => {
  //     const { asFragment } = render(<DetailsCard product={mock}></DetailsCard>);
  //     expect(asFragment()).toMatchSnapshot();
  //   });
});
