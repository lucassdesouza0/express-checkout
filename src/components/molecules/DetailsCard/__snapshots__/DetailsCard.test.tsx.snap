// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> it should render the load version of the card 1`] = `
<DocumentFragment>
  <div
    class="css-1oarku0-Card e1ss9q3t3"
  >
    <span>
      <span
        class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        style="width: 100px; height: 100px;"
      >
        ‌
      </span>
    </span>
    <div
      class="css-1lezaxk-Details e1ss9q3t2"
    >
      <span>
        <span
          class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        >
          ‌
        </span>
      </span>
      <span>
        <span
          class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        >
          ‌
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
